airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py

```diff
@@ -0,0 +1,183 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from typing import Dict, Iterable, List, Mapping
+
+import pendulum as pdm
+
+
+def make_slice(record: Dict, key_value_map: Dict) -> Dict:
+    """
+    Outputs the Dict with key:value slices for the stream.
+    EXAMPLE:
+        in: records = [{dict}, {dict}, ...],
+            key_value_map = {<slice_key_name>: <key inside record>}
+        {
+            <slice_key_name> : records.<key inside record>.value,
+            ....
+        }
+    """
+    result = {}
+    for key in key_value_map:
+        value = record[key_value_map.get(key, None)]
+        if value:
+            result.update(**{key: value})
+    return result
+
+
+def transform_data(records: List) -> Iterable[Mapping]:
```

Comment:
it is very difficult to review without examples.

Could you please for each block add an example of the current attribute and update attribute

Reply:
Move each block for separate 'attribyte' which requires 'transfrmation' to separate finction and add examples into each one.

Reply:
Done.